### ISSUE TITLE
Comparing photon states produced with CPU and GPU pipelines 

### ISSUE THREAD

**Issue by plexoos** [BNLNPPS/esi-g4ox#30]
Example command:

```
build/src/simg4ox -g esi-g4ox/geom/raindrop.gdml -m esi-g4ox/run.mac
```

Assuming the environment variables:

```
export TMP=/tmp/myname
export GEOM=mygeom
export OPTICKS_EVENT_MODE=DebugHeavy
```

the output is saved to `/tmp/myname/GEOM/mygeom/simg4ox/ALL0/`

- AXXX folder contains GPU results
- BXXX folder contains (Geant4) CPU results

```
import opticks.sysrap.sevt as s
a = s.SEvt.Load("/tmp/dmitri/GEOM/mygeom/simg4ox/ALL0/A000/")
b = s.SEvt.Load("/tmp/dmitri/GEOM/mygeom/simg4ox/ALL0/B000/")
a.f.record
b.f.record
```

**Issue:** Initial photon states are not included in the CPU output (?)

**Comment by plexoos** [comment]
I think we should have a test comparing the CPU and GPU outputs. It would help significantly with debugging and refactoring.

```python
In [16]: np.allclose(a.f.record[0,1:4], b.f.record[0,0:3])
Out[16]: True

In [17]: (a.f.record[0,1:4] == b.f.record[0,0:3]).all()
Out[17]: False
```

**Comment by buddhasystem** [comment]
> I think we should have a test comparing the CPU and GPU outputs. It would help significantly with debugging and refactoring.
> 
> ```python
> In [16]: np.allclose(a.f.record[0,1:4], b.f.record[0,0:3])
> Out[16]: True
> 
> In [17]: (a.f.record[0,1:4] == b.f.record[0,0:3]).all()
> Out[17]: False
> ```

Do you expect absolute precision? There are Python methods that allow you to define the precision in tests like this one.

**Comment by plexoos** [comment]
No, I don't expect identical results for floating point numbers. The numpy's allclose function has reasonable defaults but can be tweaked if necessary.

**Comment by buddhasystem** [comment]
Right on...

**Comment by plexoos** [comment]
A test comparing photon states produced with CPU and GPU pipelines is implemented in #31
8 out of 100 photons have different histories but the values seem to be reproducible.
